### DISCUSSION TITLE
What's next ?

### DISCUSSION THREAD

**Discussion by mindoodoo** [ratatui/ratatui#96] [Discussion]
Now that we have settled on a [logo](https://github.com/tui-rs-revival/ratatui/discussions/86), release day as well as [version number](https://github.com/tui-rs-revival/ratatui/discussions/70), we would love for everyone to share their ideas regarding what the future may hold for ratatui. At the beginning of the forking process a lot of ambitious and interesting ideas were flying around, hopefully there are many more to be pitched !

A couple points of discussion to kick things off :

- Integrating third party crates into the **base ratatui crate**
- Integrating third party crates into **another extension crate**
- Look at open feature PRs
  - #33
  - #34
  - #36
  - #25
  - **and many others...**
- #78 

Update: Those interested in discussing it live in a meeting, please mention your availability: https://calendly.com/d/y2h-b3c-n7s/ratatui-3

**Comment by nidnogg** [comment]
As a newcomer to `ratatui` in general (I've never used `tui-rs` before in any project), would anyone have any tips on where to start contributing/grokking the codebase? The current open issues seem kind of broad in a sense that I can't settle on a good candidate to look into first.

I've seen on Discord that @orhun has a pretty good template that's pending maintenance, perhaps that would make for a good one?

**Reply by orhun** [reply]
> I've seen on Discord that @orhun has a pretty good template that's pending maintenance, perhaps that would make for a good one?

Sure! We can update [the template](https://github.com/orhun/rust-tui-template) for possibly simplifying the codebase and adopt a better approach for newcomers. Come to think of it, it would be nice to move that repository to this organization. What do you think? @mindoodoo @sayanarijit

**Reply by mindoodoo** [reply]
Yeah definitely, I also think it might be worth discussing potentially separating templates from the main repo. Like making a ratatui template repo, where we can maintain orhun's and then add other ones. Food for thought.

**Reply by Nickiel12** [reply]
I definitely think that having template projects would make it easier on newcomers. And with the way that this crate is used, there can be several different design philosophies explored by using different templates. 

What I mean by that, is that some of the examples are single file programs, while others are separated into four or five files minimum. Exploring the pros and cons of each of them and explaining the thought process behind the different structuring would be beneficial for new users coming from different frameworks.

**Reply by mindoodoo** [reply]
Depending on the amount of people interested in contributing, as there will be quite a bit of work. We could create a second repo where we could maintain a set of templates for **ratatui** projects. We could also start working on a cookbook similar to the [rust cookbook,](https://rust-lang-nursery.github.io/rust-cookbook/) which I think would be quite fitting for a ui library.

**Reply by mindoodoo** [reply]
A very easy way to prioritize what to add first would be to add an entry to the readme that links to an issue to gather the main pain points beginners are encountering.

**Comment by conradludgate** [comment]
A feature we've been hoping for is https://github.com/fdehau/tui-rs/pull/552 as a lot of our users in atuin have been asking for it

**Reply by mindoodoo** [reply]
Since it was draft it wasn't imported, could you reopen it ?

**Reply by rishadbaniya** [reply]
Is there any plan to add non text stuff like Gauge into things like Table?

**Comment by mrjackwills** [comment]
Just want to ask, is it worth thinking about submitting a PR about `EnableMouseCapture`, or maybe trying to add a new feature in ratatui.

There's an open PR in Crossterm, https://github.com/crossterm-rs/crossterm/issues/640, but I don't see any movement on it.

Basically, by default, `EnableMouseCapture` captures every single mouse movement, which can have performance impacts, but it can be adjusted easily to only listen to clicks, scrolls, and/or drags

**Reply by Nickiel12** [reply]
If it isn't that hard to create a PR for, I would definitely be all for it. One of the advantages to this crate is the flexibility to choose exactly what you need, and it would be really annoying to suffer performance hits just because you needed to detect a scroll event.

**Reply by mindoodoo** [reply]
Yeah for sure, it's always worth it to submit a PR. Just make sure it's a change that's coherent with the purpose of the crate.

**Comment by gibbz00** [comment]
Just wanted to let you all know that I'm experimenting with viewport scrolling! I'm sending in an RFC as soon as I get a prototype working :)

**Comment by dzfrias** [comment]
To follow up #116, I'd love to see `tui-rs-tree-widget` considered in the meeting. I don't think I'll be able to make it to the meeting itself, but it would be cool to see it represented still!

**Reply by mindoodoo** [reply]
Will definitely be discussed. Another contributor also mentioned adding aditional widgets he had built, so we'll keep you posted !

**Comment by sayanarijit** [comment]
Those interested in discussing live in a meeting, please mention your availability: https://calendly.com/d/y2h-b3c-n7s/ratatui-3
